src/config.js

```diff
@@ -0,0 +1,115 @@
+import removeComments from './h2x/removeComments'
+import removeStyle from './h2x/removeStyle'
+import toReactNative from './h2x/toReactNative'
+import svgAttribute from './h2x/svgAttribute'
+
+const defaultConfig = {
+  bracketSpacing: undefined, // default to prettier
+  dimensions: true,
+  expandProps: true,
+  ext: 'js',
+  icon: false,
+  ids: false,
+  jsxBracketSameLine: undefined, // default to prettier
+  keepUselessDefs: false,
+  native: false,
+  precision: 3, // default to svgo
+  prettier: true,
+  ref: false,
+  replaceAttrValues: [],
+  semi: undefined, // default to prettier
+  singleQuote: undefined, // default to prettier
+  svgo: true,
+  tabWidth: undefined, // default to prettier
+  template: wrapIntoComponent,
+  title: true,
+  trailingComma: undefined, // default to prettier
+  useTabs: undefined, // default to prettier
+  viewBox: true,
+}
+
+function getH2xConfig(config) {
+  const plugins = [jsx, stripAttribute('xmlns'), removeComments, removeStyle]
+  config.replaceAttrValues.forEach(([oldValue, newValue]) => {
+    plugins.push(replaceAttrValue(oldValue, newValue))
+  })
+  if (!config.dimensions) plugins.push(removeDimensions)
+  if (config.icon) plugins.push(emSize)
+  if (config.ref) plugins.push(svgRef)
+  if (config.svgAttribute) plugins.push(svgAttribute(config.svgAttribute))
+  if (config.expandProps) plugins.push(expandProps)
+  if (config.native) plugins.push(toReactNative)
+
+  return { plugins }
+}
+
+function getSvgoConfig(config) {
+  const plugins = []
+  const svgoConfig = { plugins }
+  if (!config.title || config.icon) plugins.push({ removeTitle: true })
+  else if (config.title) plugins.push({ removeTitle: false })
+  if (config.viewBox) plugins.push({ removeViewBox: false })
+  if (config.keepUselessDefs) plugins.push({ removeUselessDefs: false })
+  if (config.ids) plugins.push({ cleanupIDs: { remove: false, minify: false } })
+  if (config.precision === 'number')
+    svgoConfig.floatPrecision = Number(svgoConfig.precision)
+  return svgoConfig
+}
+
+function getPrettierConfig(config) {
+  return {
+    semi: config.semi,
+    singleQuote: config.singleQuote,
+    tabWidth: config.tabWidth,
+    useTabs: config.useTabs,
+    trailingComma: config.trailingComma,
+    bracketSpacing: config.bracketSpacing,
+    jsxBracketSameLine: config.jsxBracketSameLine,
+  }
+}
+
+function configToOptions(config = {}) {
+  if (!config.template && config.native)
+    config.template = wrapIntoNativeComponent
+  config = { ...defaultConfig, ...config }
+
+  return {
+    svgo: config.svgo ? getSvgoConfig(config) : null,
+    h2x: getH2xConfig(config),
+    prettier: config.prettier ? getPrettierConfig(config) : null,
+    template: config.template(config),
+    ext: config.ext,
+  }
+}
+
+const explorer = cosmiconfig('svgr', {
+  sync: true,
+  cache: true,
+  rcExtensions: true,
+  transform: result => result && result.config,
+})
+
+async function resolveConfig(opts = {}) {
+  if (opts.config) {
```

Comment:
Without opts.config, it should use `process.pwd()`. In fact it is only for advanced use cases that you have to target a specific configuration.

Reply:
@neoziro the find function throws an error, should I just wrap it in a try catch if the config is not found?

Reply:
Juste use `search` (https://github.com/smooth-code/svgr/blob/next/src/plugins/svgo.js#L20) it does not throw an error.